string keypoint
uint8 corners
float64 light_x
float64 light_y
float64 light_z
float64 theta
